compiler/rustc_typeck/src/check/method/suggest.rs

```diff
@@ -348,9 +348,7 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                         let type_param = generics.type_param(param_type, self.tcx);
                         Some(self.tcx.def_span(type_param.def_id))
                     }
-                    ty::Adt(def, _) if def.did().is_local() => {
-                        tcx.def_ident_span(def.did()).map(|span| span)
-                    }
+                    ty::Adt(def, _) => Some(tcx.def_span(def.did())),
```

Comment:
this causes a lot of ui test noise and I don't think any of it is actually an improvement?

Reply:
I can revert.

Reply:
It still seems to be there?

Reply:
you also changed `def_ident_span` to `def_span`, which now points to the entire item, which is what I meant originally 😆 I didn't even notice the match guard.

Reply:
I changed it to be consistent with other places where we use `def_span` so that multiple span labels pointing at the same thing don't stagger. After a recent change `def_span` points only at the signature and not the whole item, so it is suitable to use now. I didn't move all to be `def_ident_span` because that would break the `derive` suggestions (I'm planning on reducing the span in the future, but not in this PR).

Reply:
[This is an example](https://github.com/rust-lang/rust/pull/98816/files#diff-c8eb5ab3ad01559b83f0fefba16790dbe9637b8f4cdd35086ed8c0f183e67d9dR1-R8) of what I'm saying:

```
error[E0599]: the method `get` exists for struct `Victim<'_, Self>`, but its trait bounds were not satisfied
  --> $DIR/impl-derived-implicit-sized-bound.rs:31:19
   |
LL | struct Victim<'a, T: Perpetrator + ?Sized>
   | ------------------------------------------
   | |
   | method `get` not found for this struct
   | doesn't satisfy `Victim<'_, Self>: VictimTrait`
```

Without the change the label for "method `get` not found for this struct" would point at the ident, making the visual presentation a bit less... clean.